run_test.sh

```diff
@@ -1541,6 +1543,75 @@ EOF
 echo 'Removing /usr/src/dkms_build_exclusive_test-1.0'
 rm -r /usr/src/dkms_build_exclusive_test-1.0
 
+############################################################################
+### Testing os-release detection                                         ###
+############################################################################
+echo "Backing up /etc/os-release and /usr/bin/os-release"
+# Get the contents of the os-release file. We do it first so symlinks resolve
+osrelease_contents=$(for f in /etc/os-release /usr/lib/os-release; do
+    if [ -f "$f" ]; then
+        cat "$f"
+    fi
+done)
+osrelease_cleanup() {
+    mv _etc-os-release /etc/os-release &>/dev/null || :
+    mv _usrlib-os-release /usr/lib/os-release &>/dev/null || :
+}
+trap osrelease_cleanup EXIT
+mv /etc/os-release _etc-os-release &>/dev/null || :
+mv /usr/lib/os-release _usrlib-os-release &>/dev/null || :
+
+echo "Adding the dkms_osrelease_stub-1.0 module with no os-release files (expected error)"
+run_with_expected_error 4 dkms add test/dkms_osrelease_stub-1.0 << EOF
+Error! System is missing os-release file.
+EOF
+
+echo "Creating /etc/os-release"
+cat > /etc/os-release <<< "$osrelease_contents"
+echo "Adding the dkms_osrelease_stub-1.0 module with file /etc/os-release"
+run_with_expected_output dkms add test/dkms_osrelease_stub-1.0 << EOF
+Creating symlink /var/lib/dkms/dkms_osrelease_stub/1.0/source -> /usr/src/dkms_osrelease_stub-1.0
+EOF
+run_status_with_expected_output 'dkms_osrelease_stub' << EOF
+dkms_osrelease_stub/1.0: added
+EOF
+
+echo 'Removing dkms_osrelease_stub module'
+run_with_expected_output dkms remove -k "${KERNEL_VER}" -m dkms_osrelease_stub -v 1.0 << EOF
+Module dkms_osrelease_stub 1.0 is not installed for kernel ${KERNEL_VER} (${KERNEL_ARCH}). Skipping...
+Module dkms_osrelease_stub 1.0 is not built for kernel ${KERNEL_VER} (${KERNEL_ARCH}). Skipping...
+Deleting module dkms_osrelease_stub-1.0 completely from the DKMS tree.
+EOF
+echo "Removing /usr/src/dkms_osrelease_stub-1.0"
+rm -r /usr/src/dkms_osrelease_stub-1.0
```

Comment:
I did not check for artifacts - although I think some are actually generated because without those `rm -rf` commands, the build fails

Reply:
> Can I bother you (as a separate PR down the line) to create a simple helper that prints a warning + lists the contents, and add that across all tests?

Possibly, but practically, probably not. I'm very used to BATS for testing Bash code, was that not used in this repository because it represents an extra dependency?